### PR TITLE
Update minor fixes in Readme

### PR DESCRIPTION
**Pull request by moninom1**
Description
-----------

- Update README.md to move V3.1.0 release update.

Test Steps
-----------
NA

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ NA ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by amazonKamath — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -24,7 +22,9 @@ Also see the [Submitting a bugs/feature request](https://github.com/FreeRTOS/Fre
...
+In version 3.0.0 or higher, the folder structure of FreeRTOS-Plus-TCP has changed and the files have been broken down into smaller logically separated modules. This change makes the code more modular and conducive to unit-tests. FreeRTOS-Plus-TCP V3.0.0 improves the robustness, security, and modularity of the library. Version 3.0.0 adds comprehensive unit test coverage for all lines and branches of code and has undergone protocol testing, and penetration testing by AWS Security to reduce the exposure to security vulnerabilities. Additionally, the source files have been moved to a `source` directory. This change requires modification of any existing project(s) to include the modified source files and directories. There are examples on how to use the new files and directory structure. For an example based on the Xilinx Zynq-7000, use the code in this [branch](https://github.com/aws/amazon-freertos/tree/TCPRefactorDemo) and follow these [instructions](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_xilinx.html) to build and run the demo.
```
Comment (moninom1) [inline README.md:25]: Removed - For a windows simulator based example, refer to this [demo](https://github.com/FreeRTOS/FreeRTOS/tree/TCPRefactorDemo/FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator).
As the link is no longer valid/

**Review by sirnish — DISMISSED** [review]

**Review by tony-josi-aws — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -24,7 +22,9 @@ Also see the [Submitting a bugs/feature request](https://github.com/FreeRTOS/Fre
 **Note:** All the remaining sections are generic and applies to all the versions from V3.0.0 onwards.
 
 ## Upgrading to V3.0.0 and above
```
Comment (shubnil) [inline README.md:24]: I think we should update this line. This part is valid for V3.0.0 and V3.1.0.
We can reword it as "## Upgrading to V3.0.0 and V3.1.0"

File: README.md (hunk 1 above)
Comment (shubnil) [inline README.md:25]: Same change here in line 25.

File: README.md (hunk 2 above)
Comment (moninom1) [inline README.md:24]:
```suggestion
## Upgrading to V3.0.0 and V3.1.0
```

File: README.md [hunk 3]
```diff
@@ -9,22 +9,22 @@ FreeRTOS-Plus-TCP is a lightweight TCP/IP stack for FreeRTOS. It provides a fami
...
+## Upgrading to V3.0.0 and V3.1.0
+In version 3.0.0 or higher, the folder structure of FreeRTOS-Plus-TCP has changed and the files have been broken down into smaller logically separated modules. This change makes the code more modular and conducive to unit-tests. FreeRTOS-Plus-TCP V3.0.0 improves the robustness, security, and modularity of the library. Version 3.0.0 adds comprehensive unit test coverage for all lines and branches of code and has undergone protocol testing, and penetration testing by AWS Security to reduce the exposure to security vulnerabilities. Additionally, the source files have been moved to a `source` directory. This change requires modification of any existing project(s) to include the modified source files and directories. There are examples on how to use the new files and directory structure. For an example based on the Xilinx Zynq-7000, use the code in this [branch](https://github.com/aws/amazon-freertos/tree/TCPRefactorDemo) and follow these [instructions](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_xilinx.html) to build and run the demo.
```
Comment (moninom1) [inline README.md:25]:
```suggestion
In version 3.0.0 or 3.1.0, the folder structure of FreeRTOS-Plus-TCP has changed and the files have been broken down into smaller logically separated modules. This change makes the code more modular and conducive to unit-tests. FreeRTOS-Plus-TCP V3.0.0 improves the robustness, security, and modularity of the library. Version 3.0.0 adds comprehensive unit test coverage for all lines and branches of code and has undergone protocol testing, and penetration testing by AWS Security to reduce the exposure to security vulnerabilities. Additionally, the source files have been moved to a `source` directory. This change requires modification of any existing project(s) to include the modified source files and directories. There are examples on how to use the new files and directory structure. For an example based on the Xilinx Zynq-7000, use the code in this [branch](https://github.com/aws/amazon-freertos/tree/TCPRefactorDemo) and follow these [instructions](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_xilinx.html) to build and run the demo.
```

File: README.md (hunk 2 above)
Comment (moninom1) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (moninom1) [reply README.md]: Done

**Review by shubnil — APPROVED** [review]

**Review by tony-josi-aws — APPROVED** [review]

**Pull request merged**